posts/2020-05-27-how-to-analyze-open-liberty-logs-with-splunk.adoc

```diff
@@ -0,0 +1,139 @@
+* *--log-opt splunk-format* : Specifies the message format that will get sent to Splunk’s HTTP Event Collector.
+* *WLP_LOGGING_CONSOLE_FORMAT* : An environment variable that specifies the format for the Open Liberty runtime console.
+* *WLP_LOGGING_CONSOLE_LOGLEVEL* : An environment variable which is a filter that controls the granularity of messages that go to the console.
+* *WLP_LOGGING_CONSOLE_SOURCE* : An environment variable which consists of a comma-separated list of sources that route to the console.
+
+=== Analyzing the Open Liberty container logs in Splunk
+
+In the Splunk Web Interface, navigate to *App -> Search & Reporting*.  Find the log entries aggregated from the running Open Liberty runtime container by searching with the following query, which searches for all the log entries that are from the HTTP Event Collector named `liberty` with the `line.` prefix removed from the interesting fields. The log events can be further filtered with the desired timeframe and the log information fields, as needed.
+[source, align="center"]
+----
+   source="http:liberty"  | rename line.* as *
+----
+
+image::/img/blog/blog_splunk_search.png[Splunk Search, align="center"]
+
+=== Visualizing the Open Liberty container logs in Splunk
+
+The aggregated log events from the Open Liberty Docker container can also be visualized in a collection of visualizations, which provides real-time information at a glance. I created a sample Open Liberty Splunk dashboard that can be downloaded from GitHub. This dashboard can be used with the logging data collected from JSON logging in Open Liberty. This sample dashboard visualizes message, trace, and FFDC information.
+
+. Download the sample Open Liberty Splunk dashboard from the following GitHub repository:
+https://github.com/WASdev/sample.dashboards/tree/master/Splunk%208
+
+. In Search & Reporting page, from the Splunk Web Interface, click on the *Dashboards* tab, and click *Create New Dashboard*.  In the new window that pops up, enter a title for your new dashboard and click *Create Dashboard* with the remaining fields left as-is.
+
+image::/img/blog/blog_splunk_new_dashboard.png[Splunk New Dashboard, align="center"]
+
+[start=3]
+. From the Dashboards Editor, click the *Source* tab, and replace the contents of the source editor with the contents of the sample Open Liberty Splunk Dashboard that was downloaded from Step 1, and click *Save*.
+
+image::/img/blog/blog_splunk_edit_dashboard.png[Splunk Edit Dashboard, align="center"]
+
+[start=4]
+. You can now view a collection of visualizations for each message, trace, and first failure data capture (FFDC) logging event from the Open Liberty container.  This provides real-time information about the running container. The Time Filter input can be modified to filter the log event data displayed on your dashboard, according to your desired timeframe.
```

Comment:
**change**
The Time Filter input can be modified to filter the log event data displayed on your dashboard, according to your desired timeframe.
**to**
The Time Filter input can be modified to filter the log event data displayed on your dashboard according to your desired timeframe.

(no comma needed)